bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitThermostatImpl.java

```diff
@@ -46,93 +50,93 @@
  *
  * @author Andy Lintner - Initial contribution
  */
-class HomekitThermostatImpl extends AbstractTemperatureHomekitAccessoryImpl<GroupItem>
-        implements BasicThermostat, GroupedAccessory {
+class HomekitThermostatImpl extends AbstractTemperatureHomekitAccessoryImpl<GroupItem> implements BasicThermostat {
 
-    private final String groupName;
     private final HomekitSettings settings;
-    private String currentTemperatureItemName;
-    private String heatingCoolingModeItemName;
-    private String targetTemperatureItemName;
+    @NonNull
+    private NumberItem currentTemperatureItem;
+    @NonNull
+    private StringItem targetHeatingCoolingModeItem;
+
+    @NonNull
+    private Optional<StringItem> heatingCoolingCurrentModeItem;
+
+    @NonNull
```

Comment:
Annotate the class with `@NonNullByDefault` and use `@Nullable` where applicable. See [Null annotations](https://www.openhab.org/docs/developer/guidelines.html#null-annotations).